Fix broken nosetest to pytest fixtures

aweisberg opened the pull request:
Also update README.md with a bunch of corrected/new info.

bereng commented:
You may want to mention pyenv as a hint in cases people need to have multiple python installations on the box

bschoening commented on README.md:
```diff
@@ -7,17 +7,47 @@ localhost.
-- psutil (https://pypi.org/project/psutil/)
 - Java (which version depends on the version of Cassandra you plan to use. If
-  unsure, use Java 7 as it is known to work with current versions of Cassandra).
+  unsure, use Java 8 as it is known to work with current versions of Cassandra).
```
Perhaps reference Java 11 instead here.  Java 8 is on LTS.

aweisberg replied:
Good suggestion. I updated the text. Please LMK what you think.

ekaterinadimitrova2 replied:
I think maybe 8 makes sense until we still maintain as a Community Cassandra versions like 3.x which are on Java 8. I believe we maintain them until next year or next major release.

aweisberg replied:
This comment thread is on an outdated version of the diff. I updated it to reflect that we build on 8 and support 11.

ekaterinadimitrova2 replied:
Sounds great! Thank you so much!

bschoening commented on tests/__init__.py:
```diff
@@ -8,16 +6,3 @@
 
 if sys.version_info < (3, 0):
```
Python < 3.7 is EOL

bschoening commented on README.md:
```diff
@@ -7,17 +7,47 @@ localhost.
+New to Python development?
+--------------------------
+Python has moved on since CCM started development. `pip` is the new `easy_install`,
+Python 3 is the new 2.7, and virtualenv is strongly recommended for managing multiple Python versions
```
Suggest: Python 3.7 or later is required and using virtualenv is strongly recommended for managing multiple Python versions

aweisberg commented:
@bereng @ekaterinadimitrova2 Hi! I updated this PR based on the feedback. Also added a new gotcha to the readme to help people work around AirPlay also listening on port 7k by default.

bereng commented:
+1
